edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/deployment/KubernetesDeploymentMapper.cs

```diff
@@ -237,13 +245,33 @@ List<V1EnvVar> CollectEnv(KubernetesModule module, IModuleIdentity identity)
                 .FlatMap(config => Option.Maybe(config.Mounts))
                 .Map(mounts => mounts.Where(mount => mount.Type.Equals("volume", StringComparison.InvariantCultureIgnoreCase)).ToList());
 
-            volumeMounts.Map(mounts => mounts.Select(mount => new V1Volume(KubeUtils.SanitizeDNSValue(mount.Source), emptyDir: new V1EmptyDirVolumeSource())))
+            volumeMounts.Map(mounts => mounts.Select(mount => this.GetVolume(mount)))
                 .ForEach(volumes => volumeList.AddRange(volumes));
 
             volumeMounts.Map(mounts => mounts.Select(mount => new V1VolumeMount(mount.Target, KubeUtils.SanitizeDNSValue(mount.Source), readOnlyProperty: mount.ReadOnly)))
                 .ForEach(mounts => volumeMountList.AddRange(mounts));
 
             return (volumeList, proxyMountList, volumeMountList);
         }
+
+        V1Volume GetVolume(Mount mount)
+        {
+            string name = KubeUtils.SanitizeDNSValue(mount.Source);
+            if (this.ShouldUsePvc())
+            {
+                string mountPath = mount.Target;
+                bool readOnly = mount.ReadOnly;
+                return new V1Volume(name, persistentVolumeClaim: new V1PersistentVolumeClaimVolumeSource(name, readOnly));
+            }
+            else
+            {
+                return new V1Volume(name, emptyDir: new V1EmptyDirVolumeSource());
+            }
+        }
+
+        bool ShouldUsePvc()
+        {
+            return this.persistentVolumeName.HasValue || this.storageClassName.HasValue;
```

Comment:
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1 <- this link suggests claim names can be empty and still have defaultstorageclass enabled.

Reply:
You are correct, I made a mistake translating the Preview behavior.   If the Agent gets an empty string for StorageClass, we will use that, thus signalling to k8s that we want to create a PVC against the default storage class.  You will see that change soon.